### ISSUE TITLE
Convert Date32 to native datatypes in `.to_*` functions

### ISSUE THREAD

**Issue by crmne** [pola-rs/polars#1970]
Hi @ritchie46,

Big fan of your work here!

I would like to ask, if possible, to add the option to convert dates in the native datatypes in the various `.to_*` functions, instead of the current behaviour which simply outputs the number of days since UNIX epoch 1970-01-01 (arrow's representation).

Here's an example with this data frame:

```
>>> print(days)
shape: (1, 2)
┌────────────┬────────────┐
│ first_day  ┆ last_day   │
│ ---        ┆ ---        │
│ date       ┆ date       │
╞════════════╪════════════╡
│ 2021-04-06 ┆ 2021-12-03 │
└────────────┴────────────┘
```

The current behaviour is:
```
>>> days.to_numpy()
array([[18723, 18964]], dtype=int32)

>>> days.to_json()
'{"columns":[{"name":"first_day","datatype":"Date","values":[18723]},{"name":"last_day","datatype":"Date","values":[18964]}]}'

>>> days.to_dicts()
[{'first_day': 18723, 'last_day': 18964}]
```

What I would expect is:
```
>>> days.to_numpy()
array([['2021-04-06', '2021-12-03']], dtype='datetime64[D]')

>>> days.to_json()
'{"columns":[{"name":"first_day","datatype":"Date","values":["2021-04-06"]},{"name":"last_day","datatype":"Date","values":["2021-12-03"]}]}'

>>> days.to_dicts()
[{'first_day': date(2021, 4, 6), 'last_day': date(2021, 21, 3)}]
```

Is this possible?

I'm also happy to help.

**Comment by crmne** [comment]
To be fair, the conversion in numpy is trivial, but it would be nice to have the native numpy date directly from polars:

```
>>> days.to_numpy().astype('datetime64[D]')
array([['2021-04-06', '2021-12-03']], dtype='datetime64[D]')
```

**Comment by ritchie46** [comment]
> To be fair, the conversion in numpy is trivial, but it would be nice to have the native numpy date directly from polars:
> 
> ```
> >>> days.to_numpy().astype('datetime64[D]')
> array([['2021-04-06', '2021-12-03']], dtype='datetime64[D]')
> ```

Could you make a PR for the `to_numpy` version?

The other two needs to be done from Rust.

**Comment by ritchie46** [comment]
- [x] to_list
- [x] to_dict
- [ ] to_numpy
- [ ] to_json

**Comment by alexander-beedie** [comment]
> * [x]  to_list
> * [x]  to_dict
> * [ ]  to_numpy
> * [ ]  to_json

This is a great change (just compiled the latest commits to test it) - also, looks like `df.rows()` does the same thing (returning integers back into python instead of dates/datetimes). Suitable to add to this list? Polars is advancing at a remarkable rate... :)

**Comment by ritchie46** [comment]
> > * [x]   to_list
> > * [x]   to_dict
> > * [ ]   to_numpy
> > * [ ]   to_json
> 
> This is a great change (just compiled the latest commits to test it) - also, looks like `df.rows()` does the same thing (returning integers back into python instead of dates/datetimes). Suitable to add to this list? Polars is advancing at a remarkable rate... :)

Yeap, I will add it.

**Comment by ritchie46** [comment]
`to_json` will be fixed by:  #2021
